packages/flutter/test/material/animated_icons_test.dart

```diff
@@ -8,11 +8,75 @@ import 'dart:math' as math show pi;
 
 import 'package:flutter/material.dart';
 import 'package:flutter_test/flutter_test.dart';
-import 'package:mockito/mockito.dart';
 
+import '../flutter_test_alternative.dart' show Fake;
 import '../widgets/semantics_tester.dart';
 
-class MockCanvas extends Mock implements Canvas {}
+class MockCanvas extends Fake implements Canvas {
+  Path capturedPath;
+  Paint capturedPaint;
+
+  @override
+  void drawPath(Path path, Paint paint) {
+    capturedPath = path;
+    capturedPaint = paint;
+  }
+
+  double capturedSx;
+  double capturedSy;
+
+  @override
+  void scale(double sx, [double sy]) {
+    capturedSx = sx;
+    capturedSy = sy;
+  }
+
+  final List<CanvasCall> invocations = <CanvasCall>[];
+
+  @override
+  void rotate(double radians) {
+    invocations.add(Rotate(radians));
+  }
+
+  @override
+  void translate(double dx, double dy) {
+    invocations.add(Translate(dx, dy));
+  }
+}
+
+@immutable
+abstract class CanvasCall {
```

Comment:
nit: maybe rename for clarity to `RecordedCanvasCall` and rename the subclass to `RecordedRotateCall`, etc?